test/performance/test_superdarn.py

```diff
@@ -0,0 +1,205 @@
+pr.enable()
+d = dmap.read_map()
+pr.disable()
+s = io.StringIO()
+sortby = 'cumtime'
+ps = pstats.Stats(pr, stream=s).sort_stats(sortby)
+ps.print_stats()
+print("="*60)
+print("Read map")
+print("File: ", map_file)
+print("Size: {} MB".format(os.path.getsize(map_file)/1000000))
+print("="*60)
+print(s.getvalue())
+
+dmap = pydarn.DarnRead(grid_file)
+pr = cProfile.Profile()
+pr.enable()
+d = dmap.read_grid()
+pr.disable()
+s = io.StringIO()
+sortby = 'cumtime'
+ps = pstats.Stats(pr, stream=s).sort_stats(sortby)
+ps.print_stats()
+print("="*60)
+print("Read grid")
+print("File: ", grid_file)
+print("Size: {} MB".format(os.path.getsize(grid_file)/1000000))
+print("="*60)
+print(s.getvalue())
+
+
+print("*"*60)
+print("Performance testing for DarnWrite")
+print("*"*60)
+
+
+dmap = pydarn.DmapRead(rawacf_file)
+dmap_data = dmap.read_records()
+dmap_write = pydarn.DarnWrite(dmap_data)
+pr = cProfile.Profile()
+pr.enable()
+d = dmap_write.write_rawacf("test_dmap_performance.dmap")
+pr.disable()
+s = io.StringIO()
+sortby = 'cumtime'
+ps = pstats.Stats(pr, stream=s).sort_stats(sortby)
+ps.print_stats()
+os.remove("test_dmap_performance.dmap")
+print("="*60)
+print("Write Rawacf")
+print("File: ", rawacf_file)
+print("Size: {} MB".format(os.path.getsize(rawacf_file)/1000000))
+print("="*60)
+print(s.getvalue())
+
+dmap = pydarn.DmapRead(fitacf_file)
+dmap_data = dmap.read_records()
+dmap_write = pydarn.DarnWrite(dmap_data)
+pr = cProfile.Profile()
+pr.enable()
+d = dmap_write.write_fitacf("test_dmap_performance.dmap")
+pr.disable()
+s = io.StringIO()
+sortby = 'cumtime'
+ps = pstats.Stats(pr, stream=s).sort_stats(sortby)
+ps.print_stats()
+os.remove("test_dmap_performance.dmap")
+print("="*60)
+print("Write fitacf")
+print("File: ", fitacf_file)
+print("Size: {} MB".format(os.path.getsize(fitacf_file)/1000000))
+print("="*60)
+print(s.getvalue())
+
+dmap = pydarn.DmapRead(iqdat_file)
+dmap_data = dmap.read_records()
+dmap_write = pydarn.DarnWrite(dmap_data)
+pr = cProfile.Profile()
+pr.enable()
+d = dmap_write.write_iqdat("test_dmap_performance.dmap")
+pr.disable()
+s = io.StringIO()
+sortby = 'cumtime'
+ps = pstats.Stats(pr, stream=s).sort_stats(sortby)
+ps.print_stats()
+os.remove("test_dmap_performance.dmap")
+print("="*60)
+print("Write iqdat")
+print("File: ", iqdat_file)
+print("Size: {} MB".format(os.path.getsize(iqdat_file)/1000000))
+print("="*60)
+print(s.getvalue())
+
+dmap = pydarn.DmapRead(grid_file)
+dmap_data = dmap.read_records()
+dmap_write = pydarn.DarnWrite(dmap_data)
+pr = cProfile.Profile()
+pr.enable()
+d = dmap_write.write_grid("test_dmap_performance.dmap")
+pr.disable()
+s = io.StringIO()
+sortby = 'cumtime'
+ps = pstats.Stats(pr, stream=s).sort_stats(sortby)
+ps.print_stats()
+os.remove("test_dmap_performance.dmap")
+print("="*60)
+print("Write grid")
+print("File: ", grid_file)
+print("Size: {} MB".format(os.path.getsize(grid_file)/1000000))
+print("="*60)
+print(s.getvalue())
+
+dmap = pydarn.DmapRead(map_file)
```

Comment:
These cases are pretty similar in structure. Can this duplicated code be worked into a function?

Reply:
Yes, I was thinking that since most tests are boilerplate code. However, I feel that should be a todo to refactor this code. I would like to this branch in `develop` soon rather than waiting for code refactoring.

Reply:
added it to the todo's discussion :+1: